File: ros_project/src/topic_ros/src/talker.cpp
#include <ros/ros.h>   
#include <topic_ros/gps.h>  //自定义msg产生的头文件

int main(int argc, char **argv)
{
ros::init(argc, argv, "talker");  //用于解析ROS参数，第三个参数为本节点名
ros::NodeHandle nh;    //实例化句柄，初始化node

topic_ros::gps msg;  //自定义gps消息并初始化 

ros::Publisher pub = nh.advertise<topic_ros::gps>("gps_info", 1); //创建publisher，往"gps_info"话题上发布消息
ros::Rate loop_rate(1.0);   //定义发布的频率，1HZ 
while (ros::ok())   //循环发布msg
{
 pub.publish(msg);//以1Hz的频率发布msg
 loop_rate.sleep();//根据前面的定义的loop_rate,设置1s的暂停
}
return 0;
}

File: ros_project/src/topic_ros/scripts/talker.py
#!/usr/bin/env python
#coding=utf-8
import rospy
from topic_ros.msg import gps

def talker():
    pub=rospy.Publisher("gps_info",gps,queue_size=10)
    rospy.init_node('pytalker',anonymous=True)
    rate=rospy.Rate(1)
    x=1.0
    y=2.0
    state='working'
    while not rospy.is_shutdown():
	rospy.loginfo('talker:gps:x=%f,y=%f',x,y)
	pub.publish(gps(state,x,y))
	x=1.03*x
	y=1.01*y
	rate.sleep()
if __name__ =="__main__":
    talker()
